source/opt/vector_dce.cpp

```diff
@@ -351,13 +355,19 @@ void VectorDCE::AddItemToWorkListIfNeeded(
     WorkListItem work_item, VectorDCE::LiveComponentMap* live_components,
     std::vector<WorkListItem>* work_list) {
   ir::Instruction* current_inst = work_item.instruction;
-  auto it = live_components->find(current_inst->result_id());
-  if (it == live_components->end()) {
-    live_components->emplace(
-        std::make_pair(current_inst->result_id(), work_item.components));
+
+  if (current_inst->result_id() >= live_components->size()) {
+    live_components->resize(current_inst->result_id() + 100);
```

Comment:
I think this is ok.  But it could be pathological.
Amortized constant time (in theory) would be gained by a multiplicative factor expansion, rather than additive term.

Reply:
This is implemented by an std::vector.  The size of the vector is not the same as the amount of space allocated.  I am leaving it up to the implementation of std::vector to do a multiplicative expansion when the memory allocated needs expanded.

I wanted to keep a smaller size for the vector to reduce the number of elements that need to be constructed and destructed, even if it is only ans std::unique_ptr that remains null.